test_crates/inherent_associated_pub_const_missing/new/src/lib.rs

```diff
@@ -1,10 +1,26 @@
 pub struct StructA {}
 impl StructA {
     pub const PublicConstantB: i32 = 0;
+    #[doc(hidden)]
+    pub const PublicConstantE: i32 = 0;
     pub const PublicConstantRenamedA: i32 = 0;
 }
 struct StructB {}
 impl StructB {
+    #[doc(hidden)]
     pub const PublicConstantD: i32 = 0;
     pub const PublicConstantRenamedB: i32 = 0;
 }
+#[doc(hidden)]
+pub struct StructC {}
+impl StructC {}
+#[doc(hidden)]
+pub struct StructD {}
+impl StructD {
+    pub const PublicConstantG: i32 = 0;
+}
+pub struct StructE {}
+#[doc(hidden)]
+impl StructE {
+    pub const PublicConstantI: i32 = 0;
+}
```

Comment:
Could you add these kinds of variants to the baseline as well, so that we also test whether removals of `doc(hidden)` in all these places results in the lint being triggered?

Right now, I think these tests _still_ don't target the `public_api_eligible` lines in the query I pointed out, because those lines were in the `baseline` arm of the query, whereas this code is only evaluated in the `current` arm since it's in the `new` directory in the test crate.

Also, as a bit of a nitpick: please consider adding blank lines between different items, such as between structs and impl blocks etc. Code this tight is quite difficult to read, even if `rustfmt` lets it go through.

Reply:
Sorry for the misunderstanding, so we want test cases for when we remove the const from impl block/struct that were #[doc(hidden)] or we remove the const that was #[doc(hidden)] and the lint should be triggered in such cases

Reply:
We want both kinds of test cases.

At a high level, our tests serve three purposes:
- **True-positive vs false-negative:** we expected a lint, and got a lint.
- **False-positive vs true-negative:** we didn't expect a lint, and we didn't get one.
- **One lint per problem:** we don't want to flag the same issue more than once, e.g. if a struct gets deleted, we don't _also_ want to trigger lints for each of its methods and associated constants that got deleted too.

So we want to check `#[doc(hidden)]` in all the places where it can appear, both where it's added in `new` and where it's removed in `new`, as well as where it remains unchanged between `old` and `new`.

Currently, you have the first kind of test covered well.

Removing `#[doc(hidden)]` from any of these places while also removing the constant is not a major change, and shouldn't trigger the lint. This is the second kind of test, and is not currently covered.

Making the const, its impl block, or the owning type become `#[doc(hidden)]` is a major change, but one for a different lint. (Same if the owning type got deleted altogether.) Even if the const itself is removed altogether, this lint shouldn't trigger here. This is the third kind of test, and is partially covered.

Reply:
Summary of added test cases:
1. Removing const for doc hidden struct - not triggered
2. Removing const for doc hidden impl - not triggered
3. Adding doc hidden for impl block but not removing const - not triggered
4. Adding doc hidden for impl block but removing const - triggered
5. Adding doc hidden for struct -  only `struct_now_doc_hidden` triggered
6. Removing doc hidden for struct, impl block and/or removing the const - not triggered

Reply:
Excellent, nicely done!